### ISSUE TITLE
dynamo remap collective improvement

### ISSUE THREAD

**Issue by wconstab** [pytorch/pytorch#119890]
We should update the warning message "Can't remap async version of inplace op to functional collective" to point out that users can rewrite their own code to use the functional collective.  (possibly point them to a doc with more helpful examples).  

Also, we might want to open a placeholder issue for adding dynamo support to trace work objects, and point users there to have a place to ask for help.

**Comment by awgu** [comment]
Is there any support currently for Dynamo to remap in-place collectives to functional collectives? If so, what are the conditions for it?

**Comment by yifuwang** [comment]
> Basically as long as async_op is not true dynamo remaps non functional to functional for you.

Can you remind me the reason we don't rewrite the cases where `async_op=True`? Is it to preserve user optimization?

**Comment by wconstab** [comment]
no, its bc we don't have a way to represent the 'Work' object today throughout dynamo and as an object floating around in an fx graph.  We've talked about adding the support, just haven't designed how it'd work.

**Comment by d4l3k** [comment]
@stohirov feel free! for reference this is an example of the error: https://github.com/pytorch/pytorch/blob/8e04bc90311719284546f6ff2fb721c896a30ffe/torch/distributed/_functional_collectives.py#L1692

**Comment by d4l3k** [comment]
I'm not aware of any issue, could just point them to this one or make a new one

**Comment by akshatgit07** [comment]
Hi! I’d like to take the focused error-message part of this issue as a first PyTorch contribution. I plan to update the `async_op=True` remapping error so it clearly explains that Dynamo cannot represent the returned `Work` object yet and points users toward the traceable functional-collective form, then add regression coverage for the message. I’ll keep the change scoped to this guidance and avoid the larger `Work`-object support design. Please let me know if that scope conflicts with ongoing work.
